src/Libraries/Nop.Data/Migrations/UpgradeTo460/DataMigration.cs

```diff
@@ -220,6 +222,64 @@ T getAttributeValue<T>(IList<GenericAttribute> attributes, string key, Func<stri
                     }
                 );
             }
+
+            //#5607
+            var permissionRecordTable = _dataProvider.GetTable<PermissionRecord>();
+
+            if (!permissionRecordTable.Any(prt => string.Compare(prt.SystemName, "ForceMultiFactorAuthentication", StringComparison.InvariantCultureIgnoreCase) == 0))
+            {
+                var forceMultifactorAuthenticationPermissionRecord = _dataProvider.InsertEntity(
+                    new PermissionRecord
+                    {
+                        SystemName = "ForceMultiFactorAuthentication",
+                        Name = "Security. Force multi-factor authentication",
+                        Category = "Security"
+                    }
+                );
+
+                var isForceMultifactorAuthenticationEnabled = _dataProvider.GetTable<Setting>().Any(s => string.Compare(s.Name, "MultiFactorAuthenticationSettings.ForceMultifactorAuthentication", StringComparison.InvariantCultureIgnoreCase) == 0 && 
+                string.Compare(s.Value, "True", StringComparison.InvariantCultureIgnoreCase) == 0);
+
+                if (isForceMultifactorAuthenticationEnabled)
+                {
+                    //add it to all customer roles by default except guests role
```

Comment:
Since only "Registered" role is checked [here](https://github.com/nopSolutions/nopCommerce/blob/develop/src/Presentation/Nop.Web.Framework/Mvc/Filters/ForceMultiFactorAuthenticationAttribute.cs#L83), let's also add new permission only for "Registered" role, if the setting was enabled (not for all roles, as was required first).

Reply:
@RomanovM resolve the feedback. Can you please check it?

Reply:
@fuadhasan28 The model and locales are also no longer needed, let's remove them.

Reply:
@RomanovM It has been removed. Can you please check?